### PR TITLE
Support the MD5, SHA-1, and SHA-256 hashing algorithms

### PR DESCRIPTION
**Pull request by evren-okcu**

### CODE REVIEW

File: lib/protoflow-blocks/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ publish.workspace = true
 [features]
 default = ["all", "std"]
 all = ["hash", "rand", "serde", "sysml", "tracing"]
-hash = ["dep:blake3"]
+hash = ["dep:blake3", "dep:sha2", "dep:sha1", "dep:md-5"]
```
Comment (joshuajbouw) [inline lib/protoflow-blocks/Cargo.toml:20]: Would be a good idea to have these selectable. For example, if I was to use this, I 100% only care about blake3, not the others. In that case, I only want to expect to use blake3.

File: lib/protoflow-blocks/src/blocks/hash.rs [hunk 2]
```diff
@@ -17,6 +17,9 @@ pub mod hash {
 
     pub trait HashBlocks {
         fn hash_blake3(&mut self) -> Hash;
+        fn hash_sha256(&mut self) -> Hash;
+        fn hash_sha1(&mut self) -> Hash;
+        fn hash_md5(&mut self) -> Hash;
```
Comment (joshuajbouw) [inline lib/protoflow-blocks/src/blocks/hash.rs:22]: Its a better design to have each of these algos using a trait which it implements, not part of something greater. They are part of a family, but not inherently part of the same system.

**Review by joshuajbouw — CHANGES_REQUESTED** [review]

File: lib/protoflow-blocks/Cargo.toml (hunk 1 above)
Comment (evren-okcu) [reply lib/protoflow-blocks/Cargo.toml]: done

Comment (evren-okcu) [reply lib/protoflow-blocks/Cargo.toml]: FYI @joshuajbouw I did not include other hashing algorithms to the 'all' feature.

File: lib/protoflow-blocks/src/blocks/hash.rs (hunk 2 above)
Comment (evren-okcu) [reply lib/protoflow-blocks/src/blocks/hash.rs]: done

**Review by joshuajbouw — APPROVED** [review]

File: lib/protoflow-blocks/src/types/hash_algorithm.rs [hunk 3]
```diff
@@ -26,7 +39,119 @@ impl fmt::Display for HashAlgorithm {
...
+        let mut hasher = Md5::new();
+        hasher.update(data);
+        hasher.finalize().to_vec()
+    }
+}
+
+#[cfg(feature = "hash")]
+pub struct HasherFactory;
+
+#[cfg(feature = "hash")]
```
Comment (artob) [inline lib/protoflow-blocks/src/types/hash_algorithm.rs:142]: Seems incorrect (and redundant) to enable this based on the "hash" feature. Shouldn't this whole file (Rust module) be behind any "hash-" feature being enabled?

File: lib/protoflow-blocks/src/system.rs [hunk 4]
```diff
@@ -165,9 +165,28 @@ impl HashBlocks for System {}
 
 #[cfg(feature = "hash")]
```
Comment (artob) [inline lib/protoflow-blocks/src/system.rs:166]: This feature guard won't work properly; I'd think it needs to be a logical OR between the individual "hash-" features?

**Review by artob — CHANGES_REQUESTED** [review]
LGTM generally, but the feature-gating needs work. Probably a good idea to try actually compiling with `cargo build --features=hash-sha1`, for example!

File: lib/protoflow-blocks/src/types/hash_algorithm.rs (hunk 3 above)
Comment (evren-okcu) [reply lib/protoflow-blocks/src/types/hash_algorithm.rs]: done

**Review by artob — APPROVED** [review]
LGTM now.

**Pull request merged**